### PR TITLE
Table format

### PR DESCRIPTION
**Pull request by JeffinsonDarmawan**
Added a JTextUtil ASCII table formatting for `help`, `info` and `compare` commands

### CODE REVIEW

**Comment by JeffinsonDarmawan** [comment]
image

When merging, don't forget to resync your gradle

File: src/main/java/florizz/core/FuzzyLogic.java [hunk 1]
```diff
@@ -211,75 +222,82 @@ protected static String processCommand(String userInput) throws FlorizzException
      * @throws FlorizzException if there is an issue processing the input.
      */
     protected static String splitAndMergeInput(String userInput) throws FlorizzException {
```
Comment (samuelory) [inline src/main/java/florizz/core/FuzzyLogic.java:224]: Throws unhandled exception for info and compare

**Review by samuelory — CHANGES_REQUESTED** [review]

File: src/main/java/florizz/core/Ui.java [hunk 2]
```diff
@@ -218,14 +257,23 @@ public void printFilteredFlowers(ArrayList<Flower> flowers, String filter, int p
...
-        int maxPages = (int) Math.ceil((double)lastShownList.size() / PAGE_SIZE);
-        System.out.println("Here is page " + lastPageNo + "/" + maxPages +
-                " of info regarding flowers whose name contains " + targetFlower + ":");
-        printFlowerList(true);
+    public void printFlowerInfo(ArrayList<Flower> flowers, String targetFlower, int pageNo) throws FlorizzException {
+        List<TableData> tableData = new ArrayList<>(List.of());
+        int id = 1;
+
+        for (Flower flower : flowers) {
+            tableData.add(new TableData(id, flower.getFlowerName(), flower.getColour(), flower.getOccasion().toString(),
```
[the remark this answers is not in the document]
Comment (JeffinsonDarmawan) [reply src/main/java/florizz/core/Ui.java:265]: image

Changed formatting

File: src/main/java/florizz/core/FuzzyLogic.java (hunk 1 above)
Comment (JeffinsonDarmawan) [reply src/main/java/florizz/core/FuzzyLogic.java:224]: image

Fixed Exception Handling

File: src/main/java/florizz/objects/Flower.java [hunk 3]
```diff
@@ -233,6 +236,48 @@ public String toString() {
...
+        for (String meaning : meanings){
+            meaningsString.append(meaning);
+            meaningsString.append(", ");
+        }
+        if (!(meaningsString.toString().isEmpty())) {
+            finalMeaning = meaningsString.substring(0,meaningsString.lastIndexOf(","));
+        } else {
+            finalMeaning = "-";
+        }
+        return finalMeaning;
```
Comment (samuelory) [inline src/main/java/florizz/objects/Flower.java:274]: Thanks for implementing these!

File: src/test/java/florizz/command/CompareCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        CompareCommand testCompareCommand2 = new CompareCommand("Rose", "Rose");
+        assertThrows(FlorizzException.class, () -> testCompareCommand2.execute(testList, ui));
+    }
+
+    @Test
+    void testCompareException2() { // Comparing a flower that does not exist
+        ArrayList<Bouquet> testList = new ArrayList<>();
+        Ui ui = new Ui();
+        CompareCommand testCompareCommand3 = new CompareCommand("Rose", "Tulip");
+        assertThrows(FlorizzException.class, () -> testCompareCommand3.execute(testList, ui));
```
Comment (samuelory) [inline src/test/java/florizz/command/CompareCommandTest.java:40]: great to test exception handling!

**Review by samuelory — APPROVED** [review]

**Pull request merged**